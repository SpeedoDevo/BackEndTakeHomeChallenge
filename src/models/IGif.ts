interface Data {
  title: string
  url: string
}

export default interface IGif {
  toObject(): Data
}
