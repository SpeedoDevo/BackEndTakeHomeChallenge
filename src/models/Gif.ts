import IGif from './IGif'
import IGiphy from './IGiphy'

class GiphyGif implements IGif {
  constructor(private response: IGiphy) {}

  toObject() {
    const {
      title,
      images: {
        downsized_large: { url },
      },
    } = this.response
    return { title, url }
  }
}

export default class Gif implements IGif {
  private constructor(private delegate: IGif) {}

  static fromGiphy(response: IGiphy): Gif {
    return new Gif(new GiphyGif(response))
  }

  toObject() {
    return this.delegate.toObject()
  }
}
