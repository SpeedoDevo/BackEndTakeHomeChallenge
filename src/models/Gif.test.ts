import Gif from './Gif'
import IGif from './IGif'

describe('Gif', () => {
  let gif: IGif

  beforeEach(() => {
    gif = Gif.fromGiphy({
      title: 'a gif',
      images: { downsized_large: { url: '/the-url' } },
    })
  })

  describe('toObject', () => {
    it('should match snapshot', () => {
      expect(gif.toObject()).toMatchSnapshot()
    })
  })
})
