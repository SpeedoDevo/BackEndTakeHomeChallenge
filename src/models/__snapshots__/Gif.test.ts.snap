// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif toObject should match snapshot 1`] = `
Object {
  "title": "a gif",
  "url": "/the-url",
}
`;
